pkg/supervisor/prochandle_linux.go

```diff
@@ -83,3 +123,27 @@ func (p *unixProcess) requestGracefulTermination() error {
 func (p *unixProcess) kill() error {
 	return p.process.Kill()
 }
+
+func isZombie(pid int) (bool, error) {
+	// https://man7.org/linux/man-pages/man5/proc_pid_stat.5.html
```

Comment:
Nit: Maybe it's worth adding another comment with the relevant kernel source (I had to search it to understand the fields and all)

https://github.com/torvalds/linux/blob/v4.19/fs/proc/array.c#L152

I picked 4.19 so that it's not a very recent version and people reading it don't think it's a new feature but it doesn't really matter.

Reply:
> we already depend indirectly on github.com/prometheus/procfs

Yeah, I think there's quite a few indirect dependencies dealing with procfs. I think containerd might have something into that direction, too. But I think they're all coming from a different angle to the problem: Prometheus is about monitoring, i.e. gathering data in a bunch, containerd, again, might be more concerned about security aspects when querying procfs data.